### PR TITLE
lightning(dm): export metrics

### PR DESCRIPTION
**Pull request by lance6716**
### What problem does this PR solve?

Issue Number: ref #5867 

### What is changed and how it works?

as title. But we still need update grafana to display lightning metrics

#### Tests 

 - Integration test

### Release note 

```release-note
`None`.
```

### CODE REVIEW

File: dm/loader/lightning.go
Comment (gozssky) [inline dm/loader/lightning.go:28]: I prefer a better name for `promutil2` and `promutil`. It is easy to confuse the two. (It's not a very strong request)

**Review by gozssky — APPROVED** [review]

File: dm/loader/lightning.go
Comment (lance6716) [reply dm/loader/lightning.go:28]: will rename later

File: dm/loader/lightning.go
Comment (D3Hunter) [inline dm/loader/lightning.go]: why use noop registry?

File: dm/loader/lightning.go
Comment (lance6716) [reply dm/loader/lightning.go:218]: DM changed it 

https://github.com/pingcap/tiflow/blob/30d152c0b2abcb89fd018581472e8aa09fcd674c/dm/dm/worker/metrics.go#L122

File: dm/loader/lightning.go
Comment (lance6716) [reply dm/loader/lightning.go]: If dataflow engine provides a Factory, the Factory will handle register and deregister, no other registerer is needed.

Comment (D3Hunter) [reply dm/loader/lightning.go]: maybe add a comment

**Review by D3Hunter — COMMENTED** [review]
rest lgtm after addressing previous comments

**Review by D3Hunter — APPROVED** [review]

**Comment by lance6716** [comment]
/hold

waiting to add comment to explain Factory behaviour

File: dm/loader/lightning.go
Comment (lance6716) [reply dm/loader/lightning.go]: https://github.com/pingcap/tiflow/pull/5869/commits/75bd932c9e2ed00c04283c62432a45cbb5ea577b

**Pull request merged**
